packages/modular-scripts/src/build.ts

```diff
@@ -0,0 +1,488 @@
+// While working on this file, be aware that builds
+// could be happening simultaneously across packages, so
+// try be 'thread-safe'. No state outside of functions
+
+import { JSONSchemaForNPMPackageJsonFiles as PackageJson } from '@schemastore/package';
+import { JSONSchemaForTheTypeScriptCompilerSConfigurationFile as TSConfig } from '@schemastore/tsconfig';
+
+import { promisify as prom } from 'util';
+
+import * as rollup from 'rollup';
```

Comment:
I wish that we'd had a chance to discuss this as a team prior to an implementation being written, but there's a big problem with this bundling approach for libraries. You can read more about it at https://webpack.js.org/blog/2020-12-08-roadmap-2021/#esm-library and https://material-ui.com/guides/minimizing-bundle-size/

Reply:
The first link is a 404, and it's not clear what you're suggesting in link 2. 

I've brought this up multiple times inside the team over the last couple of months, pointed to the internal PR, gave constant updates, did a walkthrough of the code, and asked for people to look at it multiple times. 

Please give specific feedback and I can address it.

Reply:
The first link is a 404 because of Service Workers, clear your cache/do a hard reset. The specific feedback is explained in detail if you navigate to the first link, but the idea is it's harmful for modular (regular use of the word modular, not Modular) libraries to be shipped as a bundle because of tree shaking issues.

Reply:
Noted as a possible tradeoff, though it would be weird if a bundler included functions in an es module that weren't referenced (I'll verify the behaviour later). imo not a deal breaker for libraries that aren't large collections of utilities (and if so, they don't have to `modular build` to build those workspaces).

Reply:
And the reason I brought up that Material-UI link is because they intentionally do not distribute their package as an ESM/CJS bundle, but rather they ship unbundled modules so that tree-shaking can work reliably.

Reply:
I'll look at preserving module structure, shouldn't be too hard.

Reply:
Going back to my original comment, I strongly suggest that you revisit the question of whether it even makes sense to bundle a library (webpack says it doesn't). Webpack are aware of the limitation that it can't produce ESM bundles, but they say that it's actually not something you'd want in the first place, hence why it isn't being prioritised.

Reply:
Yes, that's why I'm looking at preserving the module structure.

Reply:
This is updated to now preserve the module/directory structure, and doesn't make bundles.